### DISCUSSION TITLE
Tweak: replaceInternalTagLinks(node) --- Internal Tag Links need to be suppressed

### DISCUSSION THREAD

**Discussion by vgyenge6** [mvdkwast/obsidian-copy-as-html#63] [Ideas]
Hi,

I tweaked your main.js because I found the tags put in the md file are converted to <a class="tag" target="_blank" and those links navigate to a little ugly.

I wrote a new function replaceInternalTagLinks(node) and put it after replaceInternalLinks(node)
If there is an anchor in the file named the same or has an id as the tag name it will navigate to there, if there is not it will do nothing. 

```
replaceInternalTagLinks(node) {
    node.querySelectorAll("a.tag").forEach((node2) => {
	node2.setAttribute("target", "_self");
    }); 
  }
```
it needs to be called here

```
async transformHTML(element) {
    const node = element.cloneNode(true);
    node.removeAttribute("style");
    if (this.options.removeFrontMatter) {
      this.removeFrontMatter(node);
    }
    this.replaceInternalLinks(node);
    this.replaceInternalTagLinks(node);
```

I look forward to hearing your opinion. Thank you.

**Comment by mvdkwast** [comment]
A yes, tags :) I don't use in the documents I need to export. 

I think tags shouldn't be rendered as links : it doesn't seem very logical to me that they should direct to a header that the same name in the best case, or nowhere if no such header exists.

Thanks for reporting

**Comment by mvdkwast** [comment]
implemented in 53eee71b3295ad41e4b73e056ca18336b7d2d7c0

**Reply by vgyenge6** [reply]
See my thanks here https://github.com/mvdkwast/obsidian-copy-as-html/discussions/62#discussioncomment-9946766
